File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/02_Runtime_Polymorphism/javascript/vehicle.js
class Vehicle {
    start() {
        // Base implementation (empty)
    }
}

class Car extends Vehicle {
    start() {
        console.log("Starting a car");
    }
}

class Bike extends Vehicle {
    start() {
        console.log("Starting a bike");
    }
}

class Truck extends Vehicle {
    start() {
        console.log("Starting a truck");
    }
}

module.exports = { Vehicle, Car, Bike, Truck };

File: 02_Object_Oriented_Programming/05_Polymorphism_and_It's_types/02_Runtime_Polymorphism/python/vehicle.py
class Vehicle:
    def start(self):
        pass  # Base implementation (empty)


class Car(Vehicle):
    def start(self):
        print("Starting a car")


class Bike(Vehicle):
    def start(self):
        print("Starting a bike")


class Truck(Vehicle):
    def start(self):
        print("Starting a truck")

